src/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import ListContainer from './ListContainer';
+import ListItemComponent from './ListItemComponent';
+
+const redditAPI = 'https://www.reddit.com/r/';
+
+export default class SubredditPosts extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      redditPosts: []
+    };
+  }
+
+  componentDidMount() {
+    const uri = `${redditAPI}${this.props.subreddit}.json`;
+
+    fetch(uri)
+      .then(data => data.json())
+      .then(this.handlePosts)
+      .catch(err => console.error(err));
+  }
+
+  handlePosts = (posts) => {
+    const apiPosts = posts.data.children.map((post, index) => ({
+      key: index,
+      title: post.data.title,
+      media: this.getMediaFromPost(post),
+      link: post.data.url
+    }));
+
+    this.setState({
+      redditPosts: apiPosts,
+      isLoading: false
+    });
+  }
+
+  getMediaFromPost = (post) => {
+    const extension = post.data.url.split('.').pop();
+    if (post.data.hasOwnProperty('preiew') && extension === 'gif') {
+      return post.data.preview.preview.images[0].source.url;
+    }
+
+    if (extension === 'gif' || extension === 'jpg' || extension === 'jpeg') {
+      return post.data.url;
+    }
+
+    return this.props.placeholder;
+  }
+
+  render() {
```

Comment:
Always put your `render` at the bottom of your component. This is not only a recognised pattern, it's also the last thing that your component does in the lifecycle. Also from my perspective every function before the render is basically preparing the component (whether it's fetching data, setting state, computing something, manipulating structure or CSS) for the final result of the component.

Reply:
ahh yes I think that was a change left by the last pull request.